app/services/ping.rb

```diff
@@ -4,11 +4,13 @@ class Ping
 
   class << self
     def ping
-      {
+      response = {
         'foreman': {
           database: ping_database,
         },
       }.merge(plugins_ping)
+
+      response['foreman']['cache'] = ping_cache if redis_cache_store?
```

Comment:
I think you need to return `response` here.

```
[2023-10-31T02:06:54.528Z] Failure:

[2023-10-31T02:06:54.528Z] PluginTest#test_register_ping_extension [/home/jenkins/workspace/foreman-pr-unit-test/test/unit/plugin_test.rb:349]

[2023-10-31T02:06:54.528Z] Minitest::Assertion: --- expected

[2023-10-31T02:06:54.528Z] +++ actual

[2023-10-31T02:06:54.529Z] @@ -1 +1 @@

[2023-10-31T02:06:54.529Z] -{:foreman=>{:database=>{:active=>true, :duration_ms=>0}}, :foo=>{:service=>{:active=>true, :duration_ms=>0}}}

[2023-10-31T02:06:54.529Z] +nil

[2023-10-31T02:06:54.529Z] 

[2023-10-31T02:06:54.529Z] Error:

[2023-10-31T02:06:54.529Z] PluginTest#test_register_status_extension:

[2023-10-31T02:06:54.529Z] TypeError: no implicit conversion of nil into Hash

[2023-10-31T02:06:54.529Z]     app/services/ping.rb:33:in `merge'

[2023-10-31T02:06:54.529Z]     app/services/ping.rb:33:in `statuses'

[2023-10-31T02:06:54.529Z]     test/unit/plugin_test.rb:381:in `test_register_status_extension'
```

Reply:
Just two cents:

Do we want to have `cache` entry anyway or only `if redis_cache_store?` ? If we don't mind having `cache` entry anyway, we could simplify here by doing:
```
      database: ping_database,
      cache: ping_cache,
```
instead. We already have `if redis_cache_store?` check in `ping_cache`.

And if we don't want to always have `cache` entry, then we could remove the check in `ping_cache`.

Reply:
Yea -- we started discussing here (https://github.com/theforeman/foreman/pull/9793#issuecomment-1783433734). Redis will be the default so I think we are OK to omit if it's not present.